pkg/persistence/postgrespersister.go

```diff
@@ -54,16 +55,44 @@ type PostgresPersister struct {
 	version          *string
 }
 
-// PersisterVersion returns the latest version of this persistence
+// PersisterVersion returns and sets the latest version of this persistence
 func (p *PostgresPersister) PersisterVersion() (*string, error) {
 	return p.persisterVersionFromTable(postgres.VersionTableName)
 }
 
+// OldVersions returns all versions except for the most recent one for this service
+func (p *PostgresPersister) OldVersions(serviceName string) ([]string, error) {
+	return p.oldVersionsFromTable(serviceName, postgres.VersionTableName)
+}
+
 // GetTableName formats tabletype with version of this persister to return the table name
 func (p *PostgresPersister) GetTableName(tableType string) string {
 	return fmt.Sprintf("%s_%s", tableType, *p.version)
 }
 
+// DropTable drops the table with the specified tableName
+func (p *PostgresPersister) DropTable(tableName string) error {
+	_, err := p.db.Query(fmt.Sprintf("DROP TABLE %v;", tableName)) // nolint: gosec
+	return err
+}
+
+// UpdateExistenceForVersionTable updates the tableName's exists field to false in the version table
+func (p *PostgresPersister) UpdateExistenceForVersionTable(tableName string, versionNumber string) error {
```

Comment:
Perhaps considering renaming this to `UpdateExistenceFalseForVersionTable` or something like that.